base/src/model.rs

```diff
@@ -1218,7 +1225,29 @@ impl Model {
     /// * [Model::update_cell_with_number()]
     /// * [Model::update_cell_with_bool()]
     /// * [Model::update_cell_with_formula()]
-    pub fn update_cell_with_text(&mut self, sheet: u32, row: i32, column: i32, value: &str) {
+    pub fn update_cell_with_text(
+        &mut self,
+        sheet: u32,
+        row: i32,
+        column: i32,
+        value: &str,
+    ) -> Result<(), String> {
```

Comment:
Can we do the error propagation in a different PR?

That will be a fantastic learning experience by the way. Maybe you can start with one method.

You will have to add tests:

* what if the sheet is incorrect 
* what if the row, column are out of range
* is the text always valid

Let em know if you think you can do that (this is technical deb)

Reply:
Okay you mean error handling of all set functions ( leaving the merge cell part ) should be handled separately in a new PR ?

Okay I remove my last commit from here, and handle the error handling in a separate PR